### PR TITLE
DFCC 471 revisit event publishing

### PR DESCRIPTION
**Pull request by sturatcliffe**

### CODE REVIEW

File: DFC.ServiceTaxonomy.Events/Handlers/PublishToEventGridHandler.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            _eventGridContentClient = eventGridContentClient;
+            _graphSyncHelper = graphSyncHelper;
+            _publishedContentItemVersion = publishedContentItemVersion;
+            _previewContentItemVersion = previewContentItemVersion;
+            _logger = logger;
+        }
+
+        public override async Task DraftSavedAsync(SaveDraftContentContext context)
+        {
+            _logger.LogInformation("PublishToEventGridHandler::Inside DraftSavedAsync");
```
Comment (lazcool) [inline DFC.ServiceTaxonomy.Events/Handlers/PublishToEventGridHandler.cs:37]: `PublishToEventGridHandler` will be logged automatically by the generic `ILogger`, so no need to also be part of the message. Also, if you want to log the method name, could you please use `nameof`.

File: DFC.ServiceTaxonomy.Events/Handlers/PublishToEventGridHandler.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            await PublishContentEvent(context.ContentItem, ContentEventType.Unpublished);
+        }
+
+        public override async Task RemovedAsync(RemoveContentContext context)
+        {
+            _logger.LogInformation("PublishToEventGridHandler::Inside RemovedAsync");
+
+            if (context.NoActiveVersionLeft)
+            {
+                //TODO : refactor to new way where only a single event is fired
```
Comment (lazcool) [inline DFC.ServiceTaxonomy.Events/Handlers/PublishToEventGridHandler.cs:59]: This TODO can be removed.

**Comment by lazcool** [comment]
Would you mind updating the [events confluence page](https://skillsfundingagency.atlassian.net/wiki/spaces/DFC/pages/1925121253/CUI+Content+event+grid+events) with the changes, and also copy across the comment from ``.

File: DFC.ServiceTaxonomy.Events/Handlers/PublishToEventGridHandler.cs (hunk 1 above)
Comment (sturatcliffe) [reply DFC.ServiceTaxonomy.Events/Handlers/PublishToEventGridHandler.cs:37]: Ah forgot to remove these, they were just to help Simon and I trace what was going on when testing.

**Pull request merged**